charts/agent-control/charts/agent-control-deployment/templates/uninstall-job.yaml

```diff
@@ -45,18 +46,105 @@ spec:
               kubectl scale deployment --replicas=0 {{ include "newrelic.common.naming.fullname" . }}  -n {{ $.Release.Namespace }} --timeout=60s
 
               # Delete the standard resources (configmaps and secrets) managed by the agent-control
-              kubectl delete configmaps -n {{ .Release.Namespace }} -l {{ $saResourcesLabelSelector }}
-              kubectl delete secrets -n {{ .Release.Namespace }} -l {{ $saResourcesLabelSelector }}
+              kubectl delete configmaps -n {{ .Release.Namespace }} -l {{ $acResourcesLabelSelector }}
+              kubectl delete secrets -n {{ .Release.Namespace }} -l {{ $acResourcesLabelSelector }}
 
               # Delete the CRs managed by the agent-control if the corresponding CRDs exist
-              {{ range $i, $cr := $saCRList }}
+              {{ range $i, $cr := $acCRList }}
               if kubectl api-resources -o name |grep {{ $cr }}; then
-                kubectl -n {{ $.Release.Namespace }} delete {{ $cr }} -l {{ $saResourcesLabelSelector }}
+                kubectl -n {{ $.Release.Namespace }} delete {{ $cr }} -l {{ $acResourcesLabelSelector }}
               fi
               {{ end }}
       
               {{- if include "newrelic-agent-control.auth.secret.shouldRunJob" . }}
               # Delete the secrets created in the cluster
-              kubectl delete secrets -n {{ .Release.Namespace }} -l "nr-generic-secret=true"
+              kubectl delete secrets -n {{ .Release.Namespace }} -l {{ $acResourcesLabelSelector }}
```

Comment:
isn't line 50 already removing all secretes without any condition ?

Reply:
Yes..
Your question makes me wonder if the CG now tried to delete the L1 secret 🤔

Reply:
no, but it causes a warning 😕 
```
sub_thread{thread: "Garbage collector"}: executing garbage collection: garbage collector fetched resources without required labels
```

I think it is better to use two different labels